modules/database/main.tf

```
@@ -0,0 +1,74 @@
+resource "azurerm_mssql_database" "this" {
+  name                           = var.database_name
+  server_id                      = var.server_id
+  elastic_pool_id                = var.elastic_pool_id
+  collation                      = var.collation
+  enclave_type                   = var.enclave_type
+  maintenance_configuration_name = var.elastic_pool_id == null ? var.maintenance_configuration_name : null # Conflifcts with elastic pool
+  ledger_enabled                 = var.ledger_enabled
+  license_type                   = var.license_type
+  max_size_gb                    = var.max_size_gb
+  sku_name                       = var.sku_name
+  storage_account_type           = var.storage_account_type
+
+  tags = var.tags
+  dynamic "long_term_retention_policy" {
+    for_each = var.long_term_retention_policy != null ? [var.long_term_retention_policy] : []
+    content {
+      weekly_retention  = long_term_retention_policy.value.weekly_retention
+      monthly_retention = long_term_retention_policy.value.monthly_retention
+      yearly_retention  = long_term_retention_policy.value.yearly_retention
+      week_of_year      = long_term_retention_policy.value.week_of_year
+    }
+  }
+  threat_detection_policy {
+    state                      = var.threat_detection_policy.state
+    disabled_alerts            = var.threat_detection_policy.disabled_alerts
+    email_account_admins       = var.threat_detection_policy.email_account_admins
+    email_addresses            = var.threat_detection_policy.email_addresses
+    retention_days             = var.threat_detection_policy.retention_days
+    storage_account_access_key = var.threat_detection_policy.storage_account_access_key
+    storage_endpoint           = var.threat_detection_policy.storage_endpoint
+  }
```

Is this necessary when we enable threat detection at the server scope? 🤔

---

I see two options:

1. If not needed because of configuration at server scope, hardcode this to be disabled.
2. If still needed in some situations, leave it in but ensure that it's disabled by default because of server config.

---

Not totally sure, the docs are a bit vague on this. My guess is that the setting is here with serverless databases in mind. If that is the case, then we do not currently need this setting.

---

If you don't need it yet, my suggestion would be to hardcode the default values for now and add a comment that it can be parametarized when needed:
```suggestion
  # Might be irrelevant when threat detection is configured at the server level.
  # Might be relevant for serverless databases. If this is true, variables can be created.
  threat_detection_policy {
    state                = "Disabled"
    disabled_alerts      = []
    email_account_admins = "Disabled"
    email_addresses      = []
    retention_days       = 0
  }
```